tools/ARIAtools/sequential_stitching.py

```diff
@@ -0,0 +1,945 @@
+    
+    for component in comp2correct:
+        # find number of times components is referenced
+        count = np.sum(~np.isnan(np.where(conn_comp_union[:,0] == component)[0]))
+        
+        if count > 1:
+        #find the one with most points in overlap
+            max_points = np.max(conn_comp_union[np.where(conn_comp_union[:,0] == component)[0], 2])
+            # store it for forward stitching
+            ix = np.where(conn_comp_union[:,2] == max_points)[0]
+            conn_pairs = np.concatenate((conn_pairs, 
+                                         np.atleast_2d(conn_comp_union[ix,:])), axis=0)
+            
+            # store it for backward stitching
+            iy = np.where((conn_comp_union[:,0] == component) & (conn_comp_union[:,2] != max_points))[0]
+            conn_pairs_reverse = np.concatenate((conn_pairs_reverse, 
+                                                 np.atleast_2d(conn_comp_union[iy, :])), axis=0)
+
+        # If count is 1 store the connecting component pair
+        else:
+            ik = np.where(conn_comp_union[:,0] == component)[0]
+            conn_pairs = np.concatenate((conn_pairs, np.atleast_2d(conn_comp_union[ik,:])), axis=0) 
+
+    return conn_pairs, conn_pairs_reverse 
+
+def _integer_2pi_cycles(unw1 : NDArray, concom1 : NDArray, ix1 : np.float32,
+                        unw2 : NDArray, concom2 : NDArray, ix2 : np.float32,
+                        range_correction : Optional[bool] = False,
+                        print_msg : Optional[bool] = True,) -> Tuple[np.float32, np.float32, np.float32]:
+    """
+    Get mean difference of unwrapped Phase values for overlaping connected components as 2pi int cycles
+    
+    Parameters
+    ----------
+    unw1 : array
+        array containing unwrapped phase in Frame-1 (South)
+    concom1 : array
+        array containing connected components in Frame-1 (South)
+    ix1 : float
+        connected component lablein Frame-1 (South), (1...n, 0 is unrealiable)
+        
+    unw2 : array
+        array containing unwrapped phase in Frame-2 (North)
+    concom2 : array
+        array containing connected components in Frame-2 (North)
+    ix2 : float
+        connected component label in Frame-2 (North), (1...n, 0 is unrealiable)
+    range_correction: bool
+        calculate range correction due to small non 2-pi shift caused by ESD different
+        between frames
+
+    Returns
+    -------
+    diff_value : float
+        mean offset between overlapping comoponent in two frames
+    correction2pi : float
+        2pi interger cycles between overlapping comoponent in two frames
+    range_corr : float
+        correction for non 2-pi shift between overlapping comoponent in two frames 
+    """  
+    # Not sure if copy() is needed, left it here to avoid overwriting array due to numpy referencing
+    # TODO: use np.where to find the component in the data and keep overlap dimensions for comparison
```

Comment:
is this completed or still placeholder?

Reply:
It is completed, the note is for future clean-ups